R/portal.R

```diff
@@ -275,43 +276,196 @@ setup_fusions = function(short_name = "GAMBL",
+                   paste0("stable_id: ",project_name, "_allcases"), "case_list_name: Samples sequenced", "case_list_description: This is this case list that contains all samples that are profiled for mutations.",
                    paste0(c("case_list_ids:", tabseplist), collapse = " "))
 
   cat(caselistdata, sep = "\n", file = caselist_all)
 
+  #CASE LIST: BLGSP-study
+  caselist = paste0(out_dir, "case_lists/BLGSP-study.txt")
+
+  #get sample ID subset belonging to the defined case list
+  blgsp_study = get_gambl_metadata(case_set = "BLGSP-study", seq_type_filter = seq_type_filter) %>%
+    pull(Tumor_Sample_Barcode) %>%
+    intersect(these_sample_ids)
+
+  tabseplist = paste(unique(blgsp_study), collapse = "\t")
+
+  caselistdata = c(paste0("cancer_study_identifier: ", project_name),
+                   paste0("stable_id: ", project_name, "_blgsp_study"), "case_list_name: BLGSP-study", "case_list_description: For information on how this case list, refer to the documentation of GAMBLR::get_gambl_metadata.",
+                   paste0(c("case_list_ids:", tabseplist), collapse = " "))
+
+  cat(caselistdata, sep = "\n", file = caselist)
+
+  #CASE LIST: FL-study
+  caselist = paste0(out_dir, "case_lists/FL-study.txt")
+
+  #get sample ID subset belonging to the defined case list
+  fl_study = get_gambl_metadata(case_set = "FL-study", seq_type_filter = seq_type_filter) %>%
+    pull(Tumor_Sample_Barcode) %>%
+    intersect(these_sample_ids)
+
+  tabseplist = paste(unique(fl_study), collapse = "\t")
+
+  caselistdata = c(paste0("cancer_study_identifier: ", project_name),
+                  paste0("stable_id: ", project_name, "_fl_study"), "case_list_name: FL-study", "case_list_description: For information on how this case list, refer to the documentation of GAMBLR::get_gambl_metadata.",
+                  paste0(c("case_list_ids:", tabseplist), collapse = " "))
+
+  cat(caselistdata, sep = "\n", file = caselist)
+
+  #CASE LIST: DLBCL-study
+  caselist = paste0(out_dir, "case_lists/DLBCL-study.txt")
+
+  #get sample ID subset belonging to the defined case list
+  dlbcl_study = get_gambl_metadata(case_set = "DLBCL-study", seq_type_filter = seq_type_filter) %>%
+    pull(Tumor_Sample_Barcode) %>%
+    intersect(these_sample_ids)
+
+  tabseplist = paste(unique(dlbcl_study), collapse = "\t")
+
+  caselistdata = c(paste0("cancer_study_identifier: ", project_name),
+                  paste0("stable_id: ", project_name, "_dlbcl_study"), "case_list_name: DLBCL-study", "case_list_description: For information on how this case list, refer to the documentation of GAMBLR::get_gambl_metadata.",
+                  paste0(c("case_list_ids:", tabseplist), collapse = " "))
+
+  cat(caselistdata, sep = "\n", file = caselist)
+
+  #CASE LIST: FL-DLBCL-study
```

Comment:
This can be addressed separately in the next PR, but it looks like there are lots of repetitive steps here. Maybe better to wrap this into it's own helper function (non-exportable) and streamline these bits for each case list.

Reply:
I think it allows to use `@param ...` because there are some other functions in other packages that have that in the description

Reply:
Ooops, wrong thread - this comment was for another function 😄